File: CS/ToggleSwitch/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Data;
using DevExpress.Utils;
using DevExpress.XtraEditors;
using DevExpress.Skins;
using CustomEditor;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraBars.Helpers;

namespace ProgramNamespace {
    public partial class Form1 : XtraForm {
        BindingList<TestClass>  testDataSource = new BindingList<TestClass>();
        public Form1() {
            InitializeComponent();
            SkinManager.EnableFormSkins();
            DevExpress.UserSkins.BonusSkins.Register();            
            SkinHelper.InitSkinGallery(galleryControl1);
            for(int i = 0; i < 10; i++)
                testDataSource.Add(new TestClass(Convert.ToBoolean(i % 2), String.Format("item {0}", i)));
            gridControl1.DataSource = testDataSource;
            gridView1.Columns[0].ColumnEdit = new RepositoryItemToggleSwitch();
            toggleSwitch1.ToggleChanged += toggleSwitch1_ToggleChanged;
        }

        void toggleSwitch1_ToggleChanged(object sender, EventArgs e) {
            labelControl1.Text = (sender as ToggleSwitch).IsTurnedOn ? "ToggleOn" : "ToggleOFF";
        }

        class TestClass {
            public TestClass(bool v1, string v2) {
                value1 = v1;
                value2 = v2;
            }
            public bool value1 { set; get; }
            public string value2 { set; get; }

        }

       
   
    }
}



  
    

File: VB/ToggleSwitch/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.Data
Imports DevExpress.Utils
Imports DevExpress.XtraEditors
Imports DevExpress.Skins
Imports CustomEditor
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraBars.Helpers

Namespace ProgramNamespace
	Partial Public Class Form1
		Inherits XtraForm
		Private testDataSource As New BindingList(Of TestClass)()
		Public Sub New()
			InitializeComponent()
			SkinManager.EnableFormSkins()
			DevExpress.UserSkins.BonusSkins.Register()
			SkinHelper.InitSkinGallery(galleryControl1)
			For i As Integer = 0 To 9
				testDataSource.Add(New TestClass(Convert.ToBoolean(i Mod 2), String.Format("item {0}", i)))
			Next i
			gridControl1.DataSource = testDataSource
			gridView1.Columns(0).ColumnEdit = New RepositoryItemToggleSwitch()
			AddHandler toggleSwitch1.ToggleChanged, AddressOf toggleSwitch1_ToggleChanged
		End Sub

		Private Sub toggleSwitch1_ToggleChanged(ByVal sender As Object, ByVal e As EventArgs)
			labelControl1.Text = If((TryCast(sender, ToggleSwitch)).IsTurnedOn, "ToggleOn", "ToggleOFF")
		End Sub

		Private Class TestClass
			Public Sub New(ByVal v1 As Boolean, ByVal v2 As String)
				value1 = v1
				value2 = v2
			End Sub
			Private privatevalue1 As Boolean
			Public Property value1() As Boolean
				Get
					Return privatevalue1
				End Get
				Set(ByVal value As Boolean)
					privatevalue1 = value
				End Set
			End Property
			Private privatevalue2 As String
			Public Property value2() As String
				Get
					Return privatevalue2
				End Get
				Set(ByVal value As String)
					privatevalue2 = value
				End Set
			End Property

		End Class



	End Class
End Namespace






